### PR TITLE
New Min App using react-native@0.71.3 pointing to bluedot react native plugin 2.3.0

### PR DESCRIPTION
**Pull request by datkinnguyen**
Ticket: 
- Link to plugin 2.3.0 from ticket: https://bluedotinnovation.atlassian.net/browse/BD-4622
- Tap outside text field to dismiss keyboard https://bluedotinnovation.atlassian.net/browse/BD-4617

### CODE REVIEW

File: components/InitializeSdk.js [hunk 1]
```diff
@@ -85,26 +85,28 @@ export default function Initialize() {
...
-
-      {error !== null && <Text>Error authenticating {error}</Text>}
-
-      <Button
-        title="Initialize"
-        onPress={handleInitializeSDK}
-        disabled={Boolean(!projectId)}
-      />
-    </View>
+    </TouchableWithoutFeedback>
```
Comment (datkinnguyen) [inline components/InitializeSdk.js:110]: Change: add TouchableWithoutFeedback to dismiss keyboard when tapping outside of text field

File: helpers/permissionsHandler.js [hunk 2]
```diff
@@ -43,9 +43,39 @@ export const requestBluetoothPermissions = async () => {
...
   if (Platform.OS === OS.IOS) {
+    await requestBluetoothPermissions();
     await requestNotifications(['alert', 'sound']).then(({ status, settings }) => {
       console.log("Notification Permission Status: " + status);
+
+      Geolocation.setRNConfiguration({
+        skipPermissionRequests: false,
+        authorizationLevel: 'whenInUse',
+      });
+      Geolocation.requestAuthorization();
```
Comment (datkinnguyen) [inline helpers/permissionsHandler.js:77]: Change: new method `requestAllPermissions` to ask multiple permissions at once, especially for Android 13 where we have to ask both location & notification permissions. Will ask notification permission first then location. If ask at the same time, only 1 will show.

File: package.json [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
     "@react-native-async-storage/async-storage": "^1.17.10",
     "@react-native-community/geolocation": "^2.1.0",
     "@react-native-community/push-notification-ios": "^1.10.1",
     "appcenter": "4.4.5",
     "appcenter-crashes": "4.4.5",
-    "bluedot-react-native": "Bluedot-Innovation/Bluedot-React-Native-Plugin#release/2.2.1",
-    "expo": "~46.0.7",
-    "expo-splash-screen": "~0.16.1",
-    "react": "18.0.0",
+    "bluedot-react-native": "Bluedot-Innovation/Bluedot-React-Native-Plugin#dn/react-0.69-above",
```
Comment (datkinnguyen) [inline package.json:16]: will update to branch `release/2.3.0` once Plugin MR is merged. Will also update to the npmjs published one once we release plugin 2.3.0 to the public.

**Comment by datkinnguyen** [comment]
Most of the changes come from recreating the Min App with the latest react-native@0.71.3. Upgrading is way too manual + many conflicts, especially recent react-native versions changed drastically due to the new react-native architecture. We usually just re-create the Min App to have the latest app template.

Manual changes will be highlighted in comments.

File: App.js [hunk 4]
```diff
@@ -15,14 +14,12 @@ import Tempo from "./components/Tempo";
...
-    // Ask location permission
-    requestLocationPermissions();
-    requestBluetoothPermissions();
-    requestNotificationPermissions();
+    requestAllPermissions();
   }, []);
 
   return (
     <NativeRouter>
+      <StatusBar style="dark" />
```
Comment (datkinnguyen) [inline App.js:22]: Change: 
- show dark content of StatusBar
- new method to request permissions. more details further down in permissionsHandler file

File: ReleaseNotes.md [hunk 5]
```diff
@@ -1,4 +1,3 @@
 # Bluedot ReactNative Minimal Integration
 
-- Update UI to enable/disable background location updates on IOS when the app is in While in use location permissions.
-- Using [Bluedot plugin 2.2.1](https://github.com/Bluedot-Innovation/Bluedot-React-Native-Plugin) (support react-native@0.69.4)
+- Using [Bluedot plugin 2.3.0](https://github.com/Bluedot-Innovation/Bluedot-React-Native-Plugin) (support react-native@>=0.69.0).
```
Comment (datkinnguyen) [inline ReleaseNotes.md:3]: Change: update release notes, point to plugin  2.3.0

File: android/build.gradle [hunk 6]
```diff
@@ -1,59 +1,21 @@
...
 
-        if (System.properties['os.arch'] == 'aarch64') {
-            // For M1 Users we need to use the NDK 24 which added support for aarch64
-            ndkVersion = '24.0.8215888'
-        } else {
-            // Otherwise we default to the side-by-side NDK version from AGP.
-            ndkVersion = '21.4.7075529'
-        }
+        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
+        ndkVersion = "23.1.7779620"
```
Comment (adam-bluedot) [inline android/build.gradle:11]: Do we still need this if the JNI code has been removed?

**Review by adam-bluedot — APPROVED** [review]

File: android/build.gradle (hunk 6 above)
Comment (datkinnguyen) [reply android/build.gradle:11]: this file is generated by react-native when recreating the react-native app, they frequently change stuff around. 
Our plugin work is mainly on the react-native side to deal with our native Android SDK directly, not through the native module. Though, there are some places we manually need to update: Manifest (permissions, packageName...) and strings.

**Pull request merged**